### ISSUE TITLE
feature request: Storage Access Framework to write external SD

### ISSUE THREAD

**Issue by garywill** [wolpi/prim-ftpd#170]

**Comment by wolpi** [comment]
Can you provide some more detail about your request?

**Comment by garywill** [comment]
`android.permission.WRITE_MEDIA_STORAGE` give r/w permission to `media_rw` paths.

Android since 4.0 seperate `sdcard_rw` and `media_rw`. On many devices external SD, USB are mounted as `media_rw`.

And there is another `android.permission.ACCESS_ALL_EXTERNAL_STORAGE`, which I think we need to notice

**Comment by wolpi** [comment]
Trying to use both permissions gives error `Permission is only granted to system apps`. So we cannot use it :cry:

**Comment by garywill** [comment]
That's a pity. It requires SAF to write those paths. That need a lot of code, as [this app](https://github.com/ppareit/swiftp/commit/10f0d9b09c7411773c97942e42166ee492ee28d3) did

**Comment by wolpi** [comment]
This app supports SAF since version 5.0. You can select it on main screen.

**Comment by clach04** [comment]
> 
> 
> That's a pity. It requires SAF to have user selecting a path to write. That need a lot of code, as [this app](https://github.com/ppareit/swiftp/commit/10f0d9b09c7411773c97942e42166ee492ee28d3) did

@wolpi I just installed and ran a a quick test. I saw the problem @garywill described, writing to (external) SD card works but only if a directory is chosen. Picking the root of the (external) SD card results in seeing an empty directory on the ftp client.

It took me a few minutes of experimentation to work the out so this. I was about to open an issue and then found this issue. Possible options:

 * document this in the readme
 * show a dialog box/toast to say "pick a folder, not the root of the resource" when the SAF select option is requested
 * code solution @garywill mentioned

I'm OK with the first two options :-)
